compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1658,6 +1658,7 @@ impl CheckAttrVisitor<'_> {
                         E0552,
                         "unrecognized representation hint"
                     )
+                    .help("valid reprs are `C`, `align`, `packed`, `transparent`, `simd`, `i8`, `u8`, `i16`, `u16`, `i32`, `u32`, `i64`, `u64`, `i128`, `u128`, `isize`, `usize`")
```

Comment:
Ideally this list would be generated from some canonical source.

Reply:
Unfortunately we don't seem to have such a source, and there's code in another place that also replicates a lot of these checks and they all just seem to check symbols.